### ISSUE TITLE
Recent search JOB filter does not work

### ISSUE THREAD

**Issue by Felix-Braun** [zowe/vscode-extension-for-zowe#2562]
**Describe the bug**

If I search for JOBS, there are two options.
1. search job by search filter (Owner: * | Prefix: * | Status: *)
2. search job by job ID

My recent Filters, with I used are shown. If I use a recent Filter from option "job search filter" it does relay, but if I try to use a Job ID as a recent Filter, it opens the option "create job search filter" with the standard values.

**To Reproduce**

**Expected behavior**
It relays the Job ID instead of creating a new job search filter.

**Screenshots**
![job_search_filter](https://github.com/zowe/vscode-extension-for-zowe/assets/125555172/0e4217da-67e6-41b7-80c6-e9fd163bdcbe)

**Desktop (please complete the following information):**

OS: Windows 10
Zowe Explorer Version: 2.12.1
(Optional) Zowe CLI Version: 7.18.9
(Optional) Are you using Secure Credential Store? Yes

**Comment by jellypuno** [comment]
Hallo! I want to work on this. Can I take it?

**Comment by zFernand0** [comment]
Hey @jellypuno,
Nice to hear from you 😋 
Go right ahead! 🥳

**Comment by jellypuno** [comment]
**Notes:**
Error happens in ZosJobsProvider.ts; somewhere around line 722

```
        if (menuType === History) {
            const parsedHistory = this.parseJobSearchQuery(choice.label);
            if (parsedHistory.JobId) {
                return this.handleSearchByJobId(parsedHistory.JobId);
            } else {
                const quickPickPrefilledItems = this.getPopulatedPickerValues(parsedHistory);
                return this.handleEditingMultiJobParameters(quickPickPrefilledItems, node);
            }
        }
    }
```

**Comment by traeok** [comment]
Fixed and completed in #2658
